### PR TITLE
POC - Restructure video processing and add gpx, nmea and exiftool runtime support

### PR DESCRIPTION
**Pull request by malconsei**
This PR is a proof of concept covering video processing (`process` command applied to videos)

Restructuring:
* Extract common logic for track handling
* Consider all track sources the same

New features:
* Support GPX and NMEA tracks for videos
* Support  invoking exiftool internally
* Support chaining multiple parsers and/or the same parser with different options
* Support defining filename patterns for data sources based on the original video filename
  * %f: full video filename
  * %g: video filename without extension
  * %e: video extension (inclusive of .)

Examples:
```
# short format
python -m mapillary_tools.commands --verbose  process --video_geotag_source exiftool_runtime  videos/insta360/

# long format - with options
python -m mapillary_tools.commands --verbose  process --video_geotag_source {"source": "exiftool_xml", "pattern": "../%g.insv.xml"}  videos/insta360/converted/

# multiple sources
python -m mapillary_tools.commands --verbose  process --video_geotag_source {"source": "gpx", "pattern": "../%g.gpx"} --video_geotag_source {"source": "gpx", "pattern": "../%g.xml"}  videos/gopro/
```

TODO:
* Tests
* Fixes :P

### CODE REVIEW

File: mapillary_tools/commands/process.py [hunk 1]
```diff
@@ -171,6 +182,14 @@ def add_basic_arguments(self, parser: argparse.ArgumentParser):
             required=False,
             type=Path,
         )
+        group_geotagging.add_argument(
```
Comment (ptpt) [inline mapillary_tools/commands/process.py:185]: Use envvar perhaps? See FFmpeg https://github.com/mapillary/mapillary_tools/blob/main/mapillary_tools/constants.py#L20

File: mapillary_tools/commands/process.py [hunk 2]
```diff
@@ -244,31 +263,61 @@ def add_basic_arguments(self, parser: argparse.ArgumentParser):
         )
 
     def run(self, vars_args: dict):
-        if (
```
Comment (ptpt) [inline mapillary_tools/commands/process.py:247]: We can perhaps remove this special case handling for BlackVue (in next PR)

File: mapillary_tools/commands/process.py [hunk 3]
```diff
@@ -244,31 +263,61 @@ def add_basic_arguments(self, parser: argparse.ArgumentParser):
...
-        ):
-            vars_args["device_make"] = "Blackvue"
-        if (
-            "device_make" in vars_args
-            and vars_args["device_make"]
-            and vars_args["device_make"].lower() == "blackvue"
-        ):
-            vars_args["duplicate_angle"] = 360
+        if vars_args["geotag_sources"]:
+            # gpx:format=A,exif:format=B -> [{source: 'gpx', 'format': 'A'}, {'source': 'exif', format': 'B'}]
```
Comment (ptpt) [inline mapillary_tools/commands/process.py:267]: Move all the parsing logic to `process_geotag_properties.py`?

File: mapillary_tools/geotag/geotag_images_from_nmea_file.py [hunk 4]
```diff
@@ -27,7 +27,7 @@ def __init__(
         )
 
 
-def get_lat_lon_time_from_nmea(nmea_file: Path) -> T.List[geo.Point]:
+def get_lat_lon_time_from_nmea(nmea_file: Path) -> T.Sequence[geo.Point]:
```
Comment (ptpt) [inline mapillary_tools/geotag/geotag_images_from_nmea_file.py:30]: Not sure if it's expected: returning `Sequence` makes it immutable

File: mapillary_tools/video_data_extraction/extract_video_data.py [hunk 5]
```diff
@@ -0,0 +1,167 @@
...
+                "filename": file,
+                "parser": parser.parser_label,
+                "source": parser.get_geotag_source_path(),
+            }
+            if not points:
+                try:
+                    points = parser.extract_points()
+                except Exception as e:
+                    exceptions.append({"exception": e})
+                    LOG.debug(
```
Comment (ptpt) [inline mapillary_tools/video_data_extraction/extract_video_data.py:73]: LOG as warning here perhaps? I assume it'd be worthy some attention if it throws.

File: mapillary_tools/video_data_extraction/extract_video_data.py [hunk 6]
```diff
@@ -0,0 +1,167 @@
...
+
+    def _screen_sources_cardinality(self, files: T.Sequence[Path]):
+        if len(files) > 1:
+            for parser_opts in self.options["geotag_sources_options"]:
+                if "pattern" in parser_opts and "%" not in parser_opts["pattern"]:
+                    LOG.error(
+                        "Multiple video files found: Geotag source pattern for source %s must include filename placeholders",
+                        parser_opts["source"],
+                    )
+                    exit()
```
Comment (ptpt) [inline mapillary_tools/video_data_extraction/extract_video_data.py:127]: Is it the process exit? You may throw an exceptions.MapillaryUserException instead https://github.com/mapillary/mapillary_tools/blob/main/mapillary_tools/exceptions.py#L4

**Review by ptpt — COMMENTED** [review]
Nice work. This is pretty much a redo of all the `geotag_*` modules but much more clear and well organized!

File: mapillary_tools/geotag/geotag_images_from_nmea_file.py (hunk 4 above)
Comment (malconsei) [reply mapillary_tools/geotag/geotag_images_from_nmea_file.py:30]: Thanks! Reverting

File: mapillary_tools/commands/process.py (hunk 3 above)
Comment (malconsei) [reply mapillary_tools/commands/process.py:267]: Yep, that's part of "Wire them better" in the TODO list :) but I first wanted to define the options format. We can maybe discuss it tomorrow.

**Review by ptpt — APPROVED** [review]

**Pull request merged**